edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/SubscriptionChangeHandler.cs

```diff
@@ -106,6 +106,12 @@ await moduleId.Filter(i => string.Equals(i, Constants.EdgeHubModuleId))
 
         async Task HandleDirectSubscriptionChanges(Option<string> deviceId, Option<string> moduleId, List<string> subscriptionList)
         {
+            if (this.HasMixedIdentities(deviceId, moduleId, subscriptionList))
```

Comment:
This object is per client, right? Can we just cache if the client has mixed identities? 
But I suppose we cannot cache a -ve result (even if it doesn't have mixed identities now, it could in the future)

Reply:
Yes, and these subscription notifications come in a way that if the client is going to have 5 subscriptions, we get 5 notifications, always containing one item more (unless the client subscribes with a single subscription request which is not typical). So, it can happen that the first 1-2 events seem not mixed and then other identities start showing up.